internal/eventstore/handler/handler_projection.go

```diff
@@ -44,6 +44,20 @@ type Lock func(context.Context, time.Duration, ...string) <-chan error
 // Unlock releases the mutex of the projection
 type Unlock func(...string) error
 
+// Clock makes time.Now() mockable
+type Clock interface {
+	Now() time.Time
+}
+
+var _ Clock = NowFunc(nil)
+
+// NowFunc is a convenience type that implements the Clock interface
+type NowFunc func() time.Time
+
+func (n NowFunc) Now() time.Time {
+	return n()
+}
```

Comment:
looks ways too complicated to me. I have an interface to pass a function which returns me a function. doesn't feel like idiomatic go to me

Reply:
It's the same pattern the stdlib http.HandlerFunc uses 
![image](https://user-images.githubusercontent.com/12727842/227859788-28791cb2-42cf-4c06-9ebd-74aff6dd3875.png)
We already [discussed this here](https://github.com/zitadel/zitadel/pull/4779/files#r1101974217) for example.

I don't understand why it still feels so strange to you, but I accept that it needs an understanding of the pattern. I removed the Clock interface and just use a function type now, that everybody should be familiar with.